packages/agent/src/storage/walletJsonStore.ts

```diff
@@ -0,0 +1,54 @@
+import { assertAskarWallet } from '@credo-ts/askar/build/utils/assertAskarWallet'
+import type { AgentContext } from '@credo-ts/core'
+
+const store = async (context: AgentContext, id: string, value: Record<string, unknown>) => {
+  const wallet = context.wallet
+  assertAskarWallet(wallet)
+
+  wallet.withSession(async (session) => {
+    session.insert({
+      name: id,
+      value: JSON.stringify(value),
+      category: 'genericStorageRecord',
+    })
+  })
+}
+
+const replace = async (context: AgentContext, id: string, value: Record<string, unknown>) => {
+  const wallet = context.wallet
+  assertAskarWallet(wallet)
+
+  wallet.withSession(async (session) => {
+    session.replace({
+      name: id,
+      value: JSON.stringify(value),
+      category: 'genericStorageRecord',
+    })
+  })
+}
+
+const getById = async <T extends Record<string, unknown>>(context: AgentContext, id: string): Promise<T> => {
+  const wallet = context.wallet
+  assertAskarWallet(wallet)
+
+  const item = await wallet.withSession(async (session) =>
+    session.fetch({
+      name: id,
+      category: 'genericStorageRecord',
+    })
+  )
+
+  if (!item) {
+    throw new Error(`No generic item with id: ${id} found`)
+  }
+
+  const transformedValue = item.value === 'string' ? JSON.parse(item.value) : item.value
+
+  return transformedValue as T
+}
+
+export const walletJsonStore = {
+  store,
+  getById,
+  replace,
+}
```

Comment:
We have agent.genericRecords for this. No need to use askar directory.

Reply:
Ah yeah makes sense, will swap.